File: OfficeAddinWord/VB/OfficeAddinWord/ThisAddIn.vb
Imports PacificLife.Life

Public Class ThisAddIn
    Private _CmdBars As CmdBars
    Private _AppEvents As AppEvents

    Public AppEventsWatchWindow As AddinHelper.WatchWindow

    Private Sub ThisAddIn_Startup(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Startup
        PLLog.Trace("Enter", Globals.PROJECT_NAME)

        Try
            _CmdBars = New CmdBars
            _CmdBars.CreateCommandBars()

            _AppEvents = New AppEvents
            _AppEvents.Initialize()
        Catch ex As Exception
            PLLog.Error(ex, Globals.PROJECT_NAME)
        End Try

        PLLog.Trace("Exit", Globals.PROJECT_NAME)
    End Sub

    Private Sub ThisAddIn_Shutdown(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Shutdown
        PLLog.Trace("Enter", Globals.PROJECT_NAME)

        Try
            _CmdBars.RemoveCommandBars()
            _CmdBars = Nothing
            _AppEvents = Nothing
        Catch ex As Exception
            PLLog.Error(ex, Globals.PROJECT_NAME)
        End Try

        PLLog.Trace("Exit", Globals.PROJECT_NAME)
    End Sub
End Class

File: SupportTools_PowerPoint/SupportTools_PowerPoint/ThisAddIn.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Linq;
using PowerPoint = Microsoft.Office.Interop.PowerPoint;
using Office = Microsoft.Office.Core;

using PacificLife.Life;

namespace SupportTools_PowerPoint
{
    public partial class ThisAddIn
    {
        private void ThisAddIn_Startup(object sender, System.EventArgs e)
        {
            //PLLog.Trace("Enter", Common.PROJECT_NAME);
            Common.DeveloperMode = true;
            Common.WriteToDebugWindow("ThisAddIn_Startup()");
            Common.DeveloperMode = false;

            Globals.Ribbons.Ribbon.chkDisplayEvents.Checked = Common.DisplayEvents;
            Globals.Ribbons.Ribbon.chkEnableAppEvents.Checked = Common.HasAppEvents;

            try
            {
                if (Common.HasAppEvents)
                {
                    Common.AppEvents = new Events.PowerPointAppEvents();
                    Common.AppEvents.PowerPointApplication = Globals.ThisAddIn.Application;
                }

                Common.PowerPointHelper.PowerPointApplication = Globals.ThisAddIn.Application;
            }
            catch (Exception ex)
            {
                //PLLog.Error(ex, Common.PROJECT_NAME);
                Common.DeveloperMode = true;
                Common.WriteToDebugWindow(ex.ToString());
                Common.DeveloperMode = false;
            }

            //PLLog.Trace("Exit", Common.PROJECT_NAME);
        }

        private void ThisAddIn_Shutdown(object sender, System.EventArgs e)
        {
            //PLLog.Trace("Enter", Common.PROJECT_NAME);
            Common.DeveloperMode = true;
            Common.WriteToDebugWindow("ThisAddIn_Shutdown()");
            Common.DeveloperMode = false;

            try
            {
                if(Common.HasAppEvents)
                {
                    Common.AppEvents = null;
                }
            }
            catch(Exception ex)
            {
                //PLLog.Error(ex, Common.PROJECT_NAME);
                Common.DeveloperMode = true;
                Common.WriteToDebugWindow(ex.ToString());
                Common.DeveloperMode = false;
            }

            //PLLog.Trace("Exit", Common.PROJECT_NAME);
        }

        #region VSTO generated code

        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InternalStartup()
        {
            this.Startup += new System.EventHandler(ThisAddIn_Startup);
            this.Shutdown += new System.EventHandler(ThisAddIn_Shutdown);
        }

        #endregion
    }
}
